presto-main/src/main/java/com/facebook/presto/server/protocol/Query.java

```diff
@@ -507,14 +509,18 @@ private synchronized void setQueryOutputInfo(QueryExecution.QueryOutputInfo outp
 
     private synchronized URI createNextResultsUri(String scheme, UriInfo uriInfo, long nextToken)
     {
-        return uriInfo.getBaseUriBuilder()
+        UriBuilder uri = uriInfo.getBaseUriBuilder()
                 .scheme(scheme)
                 .replacePath("/v1/statement/executing")
                 .path(queryId.toString())
                 .path(String.valueOf(nextToken))
                 .replaceQuery("")
-                .queryParam("slug", slug)
-                .build();
+                .queryParam("slug", this.slug);
+        Optional<DataSize> targetResultSize = getTargetResultSize(session);
```

Comment:
how about `getTargetResultSize(session).ifPresent( targetResultSize ->  uri.queryParam("targetResultSize", targetResultSize))` ?

Reply:
It's a small point, but I didn't want the detail of whether or not the builder returns a copy to change the behavior.